### ISSUE TITLE
array_merge(): Argument #1 is not an array

### ISSUE THREAD

**Issue by rubao** [phpseclib/phpseclib#1379]
Running on PHP 7.1.24

Possible issue: the subfolder was deleted while trying to get all folder/files recursive.

Error:
```
array_merge(): Argument #1 is not an array

on vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php',
    'line' => 845,
    'function' => 'array_merge',
```

/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:827
```
function _nlist_helper($dir, $recursive, $relativeDir)
    {
        $files = $this->_list($dir, false); 

        if (!$recursive || $files === false) {
            return $files; 
        }

        $result = array();
        foreach ($files as $value) {
            if ($value == '.' || $value == '..') {
                if ($relativeDir == '') {
                    $result[] = $value;
                }
                continue;
            }
            if (is_array($this->_query_stat_cache($this->_realpath($dir . '/' . $value)))) {
                $temp = $this->_nlist_helper($dir . '/' . $value, true, $relativeDir . $value . '/');                
                $result = array_merge($result, $temp);
            } else {
                $result[] = $relativeDir . $value;
            }
        }

        return $result;
    }

```

**Comment by terrafrost** [comment]
Your PR makes it so argument #2 will always be an array whilst the error is regarding argument #1.

That said, it does look like one way you could get the error for argument #1 is by first getting the error for argument #2. eg. `array_merge([], 'a')` returns NULL and then that NULL becomes argument #1.

Anyway, I cherry picked this into the 1.0 branch and merged that into 2.0 and master:

https://github.com/phpseclib/phpseclib/commit/a7d038830c32a0004764ad552639ec63ed913ac9

Thanks!

**Comment by rubao** [comment]
Yes, the E_WARNING posted is actually the last one in the stack after `$result` is already a NULL. 
The PR is fixing the original E_WARNING.

After 
`$result = array_merge($result, $temp);` become NULL eg. `array_merge([], false)`
`$temp = $this->_nlist_helper($dir . '/' . $value, true, $relativeDir . $value . '/');` return FALSE
`if (!$recursive || $files === false) { return $files; }` return FALSE
`$files = $this->_list($dir, false); ` also return FALSE.

Thanks for merging @terrafrost !
